scripts/docker/ubuntu.14.04/Dockerfile

```diff
@@ -0,0 +1,87 @@
+#
+# Copyright (c) .NET Foundation and contributors. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#
+
+# Dockerfile that creates a container suitable to build dotnet-cli
+FROM ubuntu:14.04
+
+# Misc Dependencies for build
+RUN apt-get update && apt-get -qqy install curl unzip gettext sudo
+
+# This could become a "microsoft/coreclr" image, since it just installs the dependencies for CoreCLR (and stdlib)
+RUN echo "deb http://llvm.org/apt/trusty/ llvm-toolchain-trusty-3.6 main"  | tee /etc/apt/sources.list.d/llvm.list && \
+    curl http://llvm.org/apt/llvm-snapshot.gpg.key | apt-key add - && \
+    apt-get update && apt-get -qqy install\
+	libc6 \
```

Comment:
1. The dependency is not guaranteed to be of the same version across different versions of ubuntu, and eventually we'll need to fork this logic as older versions of ubuntu are not updated. Having that forked logic inside of a shared script instead of in the appropriate dockerfiles creates conditional logic you need to reason about to determine what exact environment the dockerfile creates when things aren't working.

2. For the same reason as above that is not guaranteed. 

3. Why do we need to figure out our dependencies set across multiple versions of ubuntu? Do we do the same comparison between fedora and ubuntu?